doc/GUIDE.md

```diff
@@ -0,0 +1,822 @@
+# Jextract Guide
+
+The jextract tool parses header (.h) files of native libraries, and generates Java code,
+called 'bindings', which  use the Foreign Function and Memory API (FFM API) under the hood,
+that can be used by a client to access the native library.
+
+Interacting with native C code through the FFM API works
+by loading a native library (e.g. a `.so`/`.dll`/`.dylib` file), which is essentially an
+archive of native functions and global variables. The user then has to look up the functions
+they want to call using a `SymbolLookup`, and finally 'link' the functions by using the
+`Linker::downcallHandle` method. Additionally, a client may need to create function pointer for
+Java functions using `Linker::upcallStub`, access global variables through the addresses
+returned by a lookup, and construct `MemoryLayout` instances for the structs they want to access.
+Jextract aims to automate many of these steps, so that a client can instead immediately start
+using a native library they are interested in.
+
+This guide shows how to run the jextract tool, and how to use the Java code that it generates.
+The samples under [`samples`](samples) direcotry are also a good source of examples.
+
+Note that at this time, jextract only supports C header files. If you have a library written
+in another language, see the section on [other languages](#other-languages).
+
+## Running Jextract
+
+A native library typically has an `include` directory which contains all the header files
```

Comment:
Here I wonder if instead of using pseudo name for paths, we should use some more real-looking names, like `/usr/include` and `/usr/lib`, which are common in many NIX systems.

Reply:
I kind of wanted to hint at the fact that all library packages should have an `include` directory. The `/usr/include` directory can seem like a special NIX system path where libraries are installed, but that's more a quirk of the package manager which chooses to put the headers of all libraries in the same directory.